connector/routingconnector/logs.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package routingconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/routingconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+)
+
+type logsConnector struct {
+	component.StartFunc
+	component.ShutdownFunc
+
+	logger *zap.Logger
+	config *Config
+	router *router[consumer.Logs]
+}
+
+func newLogsConnector(
+	set connector.CreateSettings,
+	config component.Config,
+	logs consumer.Logs,
+) (*logsConnector, error) {
+	cfg := config.(*Config)
+
+	lr, ok := logs.(connector.LogsRouter)
+	if !ok {
+		return nil, errTooFewPipelines
+	}
+
+	r, err := newRouter(
+		cfg.Table,
+		cfg.DefaultPipelines,
+		lr.Consumer,
+		set.TelemetrySettings)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &logsConnector{
+		logger: set.TelemetrySettings.Logger,
+		config: cfg,
+		router: r,
+	}, nil
+}
+
+func (c *logsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *logsConnector) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	// routingEntry is used to group plog.ResourceLogs that are routed to
+	// the same set of exporters.
+	// This way we're not ending up with all the logs split up which would cause
+	// higher CPU usage.
+	groups := make(map[consumer.Logs]plog.Logs)
+	var errs error
+
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		rlogs := ld.ResourceLogs().At(i)
+		rtx := ottlresource.NewTransformContext(rlogs.Resource())
+
+		noRoutesMatch := true
+		for _, route := range c.router.routes {
+			_, isMatch, err := route.statement.Execute(ctx, rtx)
```

Comment:
Is the error related to the data, or would it err only when the OTTL statement is wrong?

Reply:
The `Execute` method is part of the OTTL statement. Looking at [the source](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4e082f40991a2353d2994b376fbda1a95d51a96e/pkg/ottl/parser.go#L49-L66), it will error if there is an error evaluating the condition, or the function in the statement. I _think_ this would indicate a problem with the statement.

Reply:
Does this mean that this will always happen for all data points? Is it a case for a panic?